### DISCUSSION TITLE
`variables.LUA` with relative paths for local projects

### DISCUSSION THREAD

**Discussion by mrcjkb** [keplerproject/luarocks#1677] [Q&A]
Hey :wave: 

I have a project that has a Lua interpreter installed in the project-local `lua_modules` tree.
Is it possible to set `variables.LUA` in `<project_root>/.luarocks/config-5.1.lua` to point to the project-local lua interpreter?
It seems to only be accepting an absolute path, and doesn't work with environment variables like `$PWD`. Being able to do this would be very useful for using `luarocks test` both locally and in CI, with the interpreter added to the rockspec's `test_dependencies`.

**Comment by hishamhm** [comment]
wouldn't adding `luarocks config variables.LUA <path/to/lua>` to your CI script suffice?

**Reply by mrcjkb** [reply]
I'm actually using [nlua](https://luarocks.org/modules/mfussenegger/nlua) as a luarocks interpreter.
It seems like adding `lua = "~/.luarocks/bin/nlua",` to the `.busted` file, and running with `luarocks test --local` works.
But it can't seem to find the `busted.runner` module if the tree is in a project-local `lua_modules` directory for some reason.

**Reply by hishamhm** [reply]
Does the interpreter honor the LUA_PATH/CPATH variables? you can use `luarocks path` command to output the commands to set those variables (`eval $(luarocks path)`) or, alternatively you can use `luarocks path --lr-path` to get just the path names and then construct the setup that nlua needs.

**Reply by mrcjkb** [reply]
Yes, you can configure those in the `.busted` file, with `lpath` and `cpath`.
